### DISCUSSION TITLE
[feature] Ability to control dialog closing using Escape and overlay clicks.

### DISCUSSION THREAD

**Discussion by alex-oleshkevich** [tailwindlabs/headlessui#530] [Ideas]
In my case, I need to prevent dialog closing when the user clicks on overlay or presses the Esc key.

It would be nice to have the ability to control that behavior.
Thanks!

**Comment by xylophonehero** [comment]
> In my case, I need to prevent dialog closing when the user clicks on overlay or presses the Esc key.
> 
> It would be nice to have the ability to control that behavior.
> Thanks!

Was just coming here to suggest the same thing!

I have a dialog which is a full width slide out menu. The navbar is still visible as it has a higher z-index and contains the toggle button of the dialog. But when I click on the navbar, the dialog closes. I have passed an empty function to the onClose prop of the Dialog but that means I lose the Escape key close that I would like to keep.

Thanks for the great package!

**Comment by cangSDARM** [comment]
Same issue

**Comment by icep87** [comment]
Same issue.

**Comment by JavierMartinz** [comment]
Same issue, when trying to prevent the modal closing when users press `Escape`
